### PR TITLE
resolve service instance count from podinfo

### PR DESCRIPTION
**Pull request by ab77**
Change-type: minor
Signed-off-by: ab77 <anton@balena.io>

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-		"noUnusedLocals": true,
+		"noImplicitAny": true,
 		"noUnusedParameters": true,
+		"noUnusedLocals": true,
 		"outDir": "build/",
 		"preserveConstEnums": true,
 		"removeComments": true,
 		"resolveJsonModule": true,
 		"sourceMap": true,
-		"strict": true,
```
[the remark this answers is not in the document]
Comment (ab77) [reply tsconfig.json:15]: @Page- with `strict: true`, it fails on:

```
src/utils/config.ts(35,15): error TS2345: Argument of type '(key: T) => void' is not assignable to parameter of type '(value: string, index: number, array: string[]) => void'.
  Types of parameters 'key' and 'value' are incompatible.
    Type 'string' is not assignable to type 'T'.
      'string' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'string'.
```

File: src/utils/config.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	You should have received a copy of the GNU Affero General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import * as fs from 'fs';
+import * as os from 'os';
+
+export const getEnv = <T extends string>(
+	...args: string[]
```
Comment (Page-) [inline src/utils/config.ts:22]: Try
```suggestion
	...args: T[]
```
for fixing the issue with `strict`

**Comment by ab77** [comment]
@balena-ci retest

**Comment by ab77** [comment]
@balena-ci retest

**Comment by ab77** [comment]
@balena-ci retest

**Comment by ab77** [comment]
@balena-ci retest

File: tsconfig.json [hunk 3]
```diff
@@ -5,8 +5,9 @@
 		"module": "commonjs",
 		"noErrorTruncation": true,
 		"noFallthroughCasesInSwitch": true,
-		"noUnusedLocals": true,
+		"noImplicitAny": true,
```
Comment (wrboyce) [inline tsconfig.json:8]: I think this might be implied by `strict=true`?

Comment (ab77) [reply tsconfig.json:8]: I can't find a reference to it under `--strict` [here](https://www.typescriptlang.org/docs/handbook/compiler-options.html) @wrboyce

**Comment by ab77** [comment]
@balena-ci retest

File: tsconfig.json [hunk 4]
```diff
@@ -5,8 +5,8 @@
 		"module": "commonjs",
 		"noErrorTruncation": true,
 		"noFallthroughCasesInSwitch": true,
-		"noUnusedLocals": true,
 		"noUnusedParameters": true,
+		"noUnusedLocals": true,
```
Comment (wrboyce) [inline tsconfig.json:9]: NIT: these were in alphabetic order.

Comment (ab77) [reply tsconfig.json:9]: This is what `jq` spits out @wrboyce:

```
$ cat tsconfig.json | jq -r --tab .
{
        "compilerOptions": {
                "allowJs": false,
                "forceConsistentCasingInFileNames": true,
                "module": "commonjs",
                "noErrorTruncation": true,
                "noFallthroughCasesInSwitch": true,
                "noUnusedParameters": true,
                "noUnusedLocals": true,
                "outDir": "build/",
                "preserveConstEnums": true,
                "removeComments": true,
                "resolveJsonModule": true,
                "sourceMap": true,
                "strict": true,
                "skipLibCheck": true,
                "strictNullChecks": true,
                "target": "es2018"
        },
        "include": [
                "src/**/*.ts",
                "typings/**/*.d.ts"
        ],
        "exclude": [
                "node_modules"
        ]
}
```

Comment (wrboyce) [reply tsconfig.json:9]: ah, fair enough :+1:

Comment (ab77) [reply tsconfig.json]: 🙄

Comment (ab77) [reply tsconfig.json]: there is a `-S` parameter to `jq` I just learned about today @wrboyce :)

**Comment by ab77** [comment]
@balena-ci retest

**Pull request merged**